File: app/src/main/java/io/oldering/tvfoot/red/matches/state/MatchesIntent.java
package io.oldering.tvfoot.red.matches.state;

import com.google.auto.value.AutoValue;

public interface MatchesIntent {
  @AutoValue abstract class InitialIntent implements MatchesIntent {
    public static InitialIntent create() {
      return new AutoValue_MatchesIntent_InitialIntent();
    }
  }

  @AutoValue abstract class GetLastState implements MatchesIntent {
    public static GetLastState create() {
      return new AutoValue_MatchesIntent_GetLastState();
    }
  }

  @AutoValue abstract class LoadNextPageIntent implements MatchesIntent {
    public abstract int pageIndex();

    public static LoadNextPageIntent create(int pageIndex) {
      return new AutoValue_MatchesIntent_LoadNextPageIntent(pageIndex);
    }
  }
}

File: app/src/main/java/com/benoitquenaudon/tvfoot/red/app/domain/matches/state/MatchesIntent.kt
package com.benoitquenaudon.tvfoot.red.app.domain.matches.state

import com.benoitquenaudon.tvfoot.red.app.mvi.MviIntent

sealed class MatchesIntent : MviIntent {
  object InitialIntent : MatchesIntent()

  object RefreshIntent : MatchesIntent()

  object GetLastState : MatchesIntent()

  data class LoadNextPageIntent(val pageIndex: Int) : MatchesIntent()

  object ClearFilters : MatchesIntent()

  data class ToggleFilterIntent(val tagName: String) : MatchesIntent()
}
